### PR TITLE
style: use ruff for formatting and linting

### PR DESCRIPTION
**Pull request by DaniBodor**
This replaces the black formatter and prospector linter

### CODE REVIEW

**Comment by DaniBodor** [comment]
@psomhorst, not sure what you think the best way to go about this is, without creating too many conflicts.

For now I (largely) only did the changes that can be run automatically by the formatter/linter. Should I manually fix all the rest already as well? Seems a bit overkill as I believe that many of the modules will be removed (right?).
Alternatively, should I remove even these changes, and only do the settings, and then as we work on each individual file, we fix it at that moment?

**Comment by DaniBodor** [comment]
@psomhorst , I now adapted this PR to just have the ruff linting and formatting settings, but didn't apply any of them yet. After merging this into any working branch and installing ruff in the environment, it should format on save and the linter can be run using `ruff <path>` to only run it on specific files or folders. The `--fix` option can be used to let ruff auto-fix the issues that it can.

File: eitprocessing/plotting/plot.py
Comment (psomhorst) [inline eitprocessing/plotting/plot.py:1]: I guess this one shouldn't be in this PR? But it's not an issue that it's here.

**Review by psomhorst — APPROVED** [review]
This looks great! I think the approach is good.

**Pull request merged**